x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/settings.tsx

```diff
@@ -95,6 +106,74 @@ export const SettingsPage: React.FC<Props> = memo(({ packageInfo }: Props) => {
     kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${name}`,
   });
 
+  const { notifications } = useStartServices();
+
+  const shouldShowKeepPoliciesUpToDateSwitch = useMemo(() => {
+    const packages = [...DEFAULT_PACKAGES, ...AUTO_UPDATE_PACKAGES];
+
+    const packageNames = uniq(packages.map((pkg) => pkg.name));
+
+    return packageNames.includes(name);
+  }, [name]);
+
+  const [keepPoliciesUpToDateSwitchValue, setKeepPoliciesUpToDateSwitchValue] = useState<boolean>(
+    keepPoliciesUpToDate ?? false
+  );
+
+  const handleKeepPoliciesUpToDateSwitchChange = useCallback(() => {
+    const saveKeepPoliciesUpToDate = async () => {
+      try {
+        setKeepPoliciesUpToDateSwitchValue((prev) => !prev);
+
+        await sendUpdatePackage(`${packageInfo.name}-${packageInfo.version}`, {
+          keepPoliciesUpToDate: !keepPoliciesUpToDateSwitchValue,
+        });
+
+        notifications.toasts.addSuccess({
+          title: toMountPoint(
+            <FormattedMessage
+              id="xpack.fleet.integrations.integrationSaved"
+              defaultMessage="Integration settings saved"
+            />
+          ),
+          text: toMountPoint(
+            !keepPoliciesUpToDateSwitchValue ? (
+              <FormattedMessage
+                id="xpack.fleet.integrations.keepPoliciesUpToDateEnabledSuccess"
+                defaultMessage="Fleet will automatically keep integration policies up to date for {title}"
+                values={{ title }}
+              />
+            ) : (
+              <FormattedMessage
+                id="xpack.fleet.integrations.keepPoliciesUpToDateDisabledSuccess"
+                defaultMessage="Fleet will not automatically keep integration policies up to date for {title}"
+                values={{ title }}
+              />
+            )
+          ),
+        });
+      } catch (error) {
+        notifications.toasts.addError(error, {
+          title: i18n.translate('xpack.fleet.integrations.integrationSavedError', {
```

Comment:
there is two different pattern here for the notification one using `i18n.translate` and one using `toMountPoint(,<Formatted...` is there any particular reason for that?

Reply:
I believe the Error toasts accept a string and the general toasts accept a component. I'm honestly not sure about the `toMountPoint` helper I just saw it used in other similar toasts.

Reply:
I see what you mean now. I will make these both use `i18n.translate` - that seems a little more consistent.